diff --git a/node_modules/react-native-call-detection/android/build.gradle b/node_modules/react-native-call-detection/android/build.gradle
index 8903c5f..e92452d 100644
--- a/node_modules/react-native-call-detection/android/build.gradle
+++ b/node_modules/react-native-call-detection/android/build.gradle
@@ -1,8 +1,8 @@
 apply plugin: 'com.android.library'
 
-def DEFAULT_COMPILE_SDK_VERSION     = 28
-def DEFAULT_BUILD_TOOLS_VERSION     = "28.0.3"
-def DEFAULT_TARGET_SDK_VERSION      = 27
+def DEFAULT_COMPILE_SDK_VERSION     = 34
+def DEFAULT_MIN_SDK_VERSION         = 24
+def DEFAULT_TARGET_SDK_VERSION      = 34
 
 def safeExtGet(prop, fallback) {
     rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
@@ -10,25 +10,22 @@ def safeExtGet(prop, fallback) {
 
 android {
     compileSdkVersion safeExtGet('compileSdkVersion', DEFAULT_COMPILE_SDK_VERSION)
-    buildToolsVersion safeExtGet('buildToolsVersion', DEFAULT_BUILD_TOOLS_VERSION)
 
     defaultConfig {
-        minSdkVersion 16
+        minSdkVersion safeExtGet('minSdkVersion', DEFAULT_MIN_SDK_VERSION)
         targetSdkVersion safeExtGet('targetSdkVersion', DEFAULT_TARGET_SDK_VERSION)
         versionCode 1
         versionName "1.0"
     }
 }
 
-repositories{
-    jcenter()
-    maven {
-        url './libs'
-    }
+repositories {
+    google()
+    mavenCentral()
 }
 
 dependencies {
     api fileTree(dir: 'libs', include: ['*.jar'])
-    api 'com.android.support:appcompat-v7:23.0.1'
-    api 'com.facebook.react:react-native:+'
+    implementation 'androidx.appcompat:appcompat:1.7.0'
+    implementation "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
 }
